activerecord/test/cases/tasks/database_tasks_test.rb

```diff
@@ -152,23 +156,42 @@ def test_creates_configurations_with_blank_hosts
 
       ActiveRecord::Tasks::DatabaseTasks.create_all
     end
+
+    def test_creates_all_databases_under_env
+      @configurations["development"] = {
+        "primary"  => { "database" => "primary-db", "adapter" => "sqlite3" },
+        "readonly" => { "database" => "readonly-db", "adapter" => "sqlite3" },
+      }
+      ActiveRecord::Tasks::DatabaseTasks.expects(:create).
+        with("database" => "primary-db", "adapter" => "sqlite3")
+
+      ActiveRecord::Tasks::DatabaseTasks.expects(:create).
+        with("database" => "readonly-db", "adapter" => "sqlite3")
+
+      ActiveRecord::Tasks::DatabaseTasks.create_all
+    end
   end
 
   class DatabaseTasksCreateCurrentTest < ActiveRecord::TestCase
     def setup
+      @prev_config = ActiveRecord::Base.configurations
       @configurations = {
-        "development" => { "database" => "dev-db" },
-        "test"        => { "database" => "test-db" },
-        "production"  => { "database" => "prod-db" }
+        "development" => { "database" => "dev-db", "adapter" => "sqlite3" },
```

Comment:
I think it's because we only check for the `adapter` key at the moment. Finding a more accurate way to find an actual config could fix the need to change this.